Modules/_hashopenssl.c

```diff
@@ -329,24 +345,37 @@ EVP_repr(EVPobject *self)
     return PyUnicode_FromFormat("<%U HASH object @ %p>", self->name, self);
 }
 
-#if HASH_OBJ_CONSTRUCTOR
+/*[clinic input]
+_hashlib.HASH.__init__ as EVP_tp_init
+
+    name as name_obj: object
+    string as data_obj: object = NULL
```

Comment:
Does passing an empty bytes object have the same effect as omitting the second argument? If yes, I suggest to use `b''` as Python default.